src/substrate/filters/enricher.ts

```diff
@@ -455,6 +455,110 @@ export async function Enrich(
         };
       }
 
+      case EventKind.TreasuryBountyProposed: {
+        const [ bountyIndex ] = event.data as unknown as [ BountyIndex ] & Codec;
+        const bountyOpt = await api.query.treasury.bounties<Option<Bounty>>(bountyIndex);
+        if (!bountyOpt.isSome) {
+          throw new Error(`could not fetch treasury proposal index ${+bountyIndex}`);
+        }
+        const bounty = bountyOpt.unwrap();
+        return {
+          data: {
+            kind,
+            bountyIndex: +bountyIndex,
+            proposer: bounty.proposer.toString(),
+            value: bounty.value.toString(),
+            fee: bounty.fee.toString(),
+            curatorDeposit: bounty.curatorDeposit.toString(),
+            bond: bounty.bond.toString(),
+            status: {
```

Comment:
Oof -- why are we storing the entire status here, when I assume at time of "BountyProposed" we know what state it's in?

Reply:
You're right! We can also always verify the status by querying for the bounty on-chain.